lib/OpenCloud/ObjectStore/Resource/DataObject.php

```diff
@@ -293,6 +300,26 @@ public function getEtag()
     {
         return $this->etag ? : $this->content->getContentMd5();
     }
+    
+    /**
+     * @param $manifest
+     * @return $this
+     */
+    public function setManifest($manifest)
```

Comment:
I strongly recommend making this method `protected` instead of `public` for the following reasons:
* By making it `public` a user might call it thinking they are modifying the manifest on the remote service object, when in fact, they would only be changing it in the local, in-memory, object.
* At the moment this method is only being called by other methods within this class, most notably the `symlink` method which reflects an actual use case. In the future, if there is a use case to set the manifest in just the local, in-memory object, we can increase its visibility to `public` from `protected`.